### PR TITLE
fix(bundler/nsis): write installer templates UTF16LE encoded, closes #7036

### LINKED ISSUE #7036: Bundling failed on Windows when the package product name contains Chinese characters[bug]
### Describe the bug

Bundling failed on Windows when the package product name contains Chinese character
image

### Reproduction

_No response_

### Expected behavior

Package successfully

### Platform and versions

```text
[✘] Environment
    - OS: Windows 10.0.17763 X64
    ✔ WebView2: 112.0.1722.39
    ✘ Couldn't detect Visual Studio or Visual Studio Build Tools. Download from https://aka.ms/vs/17/release/vs_BuildTools.exe    ✔ rustc: 1.66.1 (90743e729 2023-01-10)
    ✔ Cargo: 1.66.1 (ad779e08b 2023-01-10)
    ✔ rustup: 1.25.1 (bb60b1e89 2022-07-12)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.14.2
    - pnpm: 7.26.3
    - yarn: 1.22.19
    - npm: 8.6.0

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.1
    - tao [RUST]: 0.16.0
    - @tauri-apps/api [NPM]: 1.3.0
    - @tauri-apps/cli [NPM]: 1.3.1

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist-pro
    - devPath: http://localhost:3000/
    - framework: Vue.js
    - bundler: Rollup
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Review by wusyong — DISMISSED** [review]

**Review by lucasfernog — DISMISSED** [review]

**Review by lucasfernog — DISMISSED** [review]

File: .changes/nsis-encoding.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri-bundler': 'patch'
```
Comment (lucasfernog) [inline .changes/nsis-encoding.md]:
```suggestion
'tauri-bundler': 'patch:bug'
```

Comment (amrbashir) [inline .changes/nsis-encoding.md]:
```suggestion
'tauri-bundler': 'patch:bug'
```

Comment (lucasfernog) [reply .changes/nsis-encoding.md]: HAHAHAHAHAHAHAHAHAHAHAHAHAHAHAHA

Comment (amrbashir) [reply .changes/nsis-encoding.md]: lol

Comment (lucasfernog) [reply .changes/nsis-encoding.md]: YOU WILL NEVER BE AS FAST

Comment (amrbashir) [reply .changes/nsis-encoding.md]: fast lucas goes vroom vroom

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**